src/app/sub/mount_point.rs

```diff
@@ -0,0 +1,41 @@
+use std::{ffi::OsStr, path::Path};
+
+pub fn find_mount_point<'a>(
+    path: &Path,
+    mount_points: impl IntoIterator<Item = &'a Path>,
+) -> Option<&'a Path> {
+    mount_points
+        .into_iter()
+        .filter(|mnt| path.starts_with(&*mnt.to_string_lossy()))
+        .max_by_key(|mnt| AsRef::<OsStr>::as_ref(mnt).len()) // Mount points can be nested in each other
+}
+
+#[cfg(test)]
+mod tests {
+    use super::find_mount_point;
+    use std::path::Path;
```

Comment:
We prefer `pretty_assertions`.

```suggestion
    use super::find_mount_point;
    use std::path::Path;
    use pretty_assertions::assert_eq;
```